src/datomish_user_agent_service/api.cljc

```diff
@@ -278,41 +278,32 @@
       (map (fn [[page url title excerpt]]
              {:url url :title title :excerpt excerpt :snippet "" :lastVisited nil})))))
 
-;; TODO: use since.
-;; TODO: return lastVisited, sort by lastVisited.
-;; TODO: support snippet extraction.
 (defn <pages-matching-string [db string {:keys [limit since]
                                          :or {:limit 10}}]
-  (let [string (str "*" string "*")] ;; Wildcard match.  TODO: escape string properly.
+  (let [string (str "*" string "*") ;; Wildcard match.  TODO: escape string properly.
+
+        ;; TODO: extract matching snippet.
+        ;; TODO: return lastVisited, order by lastVisited.
+        ;; TODO: use since, if present.
+        query
+        '[:find [?url (max ?title)]   ; Only take one title, so we're unique on URL.
```

Comment:
Yes, it's an aggregate on the value itself. Strictly speaking it's unnecessary, because `:page/title` is a unique attribute…

Still in the TODO is to expand this query. Supporting pull attributes would make this easier: it helps turn queries into a conceptual pair of a 'select' query to find things, and an 'expand' pull expression to flesh out.